jax/lax/lax.py

```diff
@@ -2309,20 +2309,11 @@ def _conv_general_dilated_shape_rule(
            "must be a positive integer, got {}.")
     raise ValueError(msg.format(batch_group_count))
   lhs_batch_count = lhs.shape[dimension_numbers.lhs_spec[0]]
+  
   if lhs_batch_count % batch_group_count != 0:
     msg = ("conv_general_dilated batch_group_count must divide lhs batch "
            "dimension size, but {} does not divide {}.")
     raise ValueError(msg.format(batch_group_count, lhs_batch_count))
-  if rhs.shape[dimension_numbers.rhs_spec[0]] % feature_group_count:
```

Comment:
This should be testing the `batch_group_count`.

Reply:
(i.e., the LHS batch dimension should be divisible by the batch_group_count).

Reply:
Is that not sufficiently covered by the condition above this?

Reply:
Err sorry you are right. The test we want is:
https://github.com/tensorflow/tensorflow/blob/f653ab8bb3910959c8189f163e5626df61ff0eab/tensorflow/compiler/xla/service/shape_inference.cc#L1736

i.e., `kernel_output_features` is divisible by `batch_group_count`

Reply:
(The Python here is ultimately testing the shape rules as XLA, since the JAX `conv` is almost 1:1 with the XLA operator. However, we prefer to catch shape errors early in Python rather than during XLA lowering.)

Reply:
Ah, thanks for the context.

Reply:
Done.